casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -51,34 +51,32 @@ class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer:
       hash: StateHash,
       terms: Seq[InternalProcessedDeploy],
       time: Option[Long] = None
-  )(
-      implicit scheduler: Scheduler
-  ): Either[(Option[Deploy], Failed), StateHash] = {
-    val runtime = runtimeContainer.take()
-    unsafeSetTimestamp(time, runtime)
-    val result = replayEval(terms, runtime, hash)
-    runtimeContainer.put(runtime)
-    result
-  }
+  ): Task[Either[(Option[Deploy], Failed), StateHash]] =
+    for {
+      runtime <- Task.delay(runtimeContainer.take())
```

Comment:
do we have a JIRA for getting switching this from SyncVar?

Reply:
IDK.

Reply:
We have now: https://rchain.atlassian.net/browse/RHOL-1034